File: sandbox/java/src/main/java/com/dropbox/sign_sandbox/ApiAppUpdateExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.*;
import com.dropbox.sign.auth.*;
import com.dropbox.sign.JSON;
import com.dropbox.sign.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class ApiAppUpdateExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");
        // ((HttpBearerAuth) config.getAuthentication("oauth2")).setBearerToken("YOUR_ACCESS_TOKEN");

        var oauth = new SubOAuth();
        oauth.callbackUrl("https://example.com/oauth");
        oauth.scopes(List.of (
            SubOAuth.ScopesEnum.BASIC_ACCOUNT_INFO,
            SubOAuth.ScopesEnum.REQUEST_SIGNATURE
        ));

        var whiteLabelingOptions = new SubWhiteLabelingOptions();
        whiteLabelingOptions.primaryButtonColor("#00b3e6");
        whiteLabelingOptions.primaryButtonTextColor("#ffffff");

        var apiAppUpdateRequest = new ApiAppUpdateRequest();
        apiAppUpdateRequest.callbackUrl("https://example.com/dropboxsign");
        apiAppUpdateRequest.name("New Name");
        apiAppUpdateRequest.domains(List.of (
            "example.com"
        ));
        apiAppUpdateRequest.customLogoFile(new File("CustomLogoFile.png"));
        apiAppUpdateRequest.oauth(oauth);
        apiAppUpdateRequest.whiteLabelingOptions(whiteLabelingOptions);

        try
        {
            var response = new ApiAppApi(config).apiAppUpdate(
                "0dd3b823a682527788c4e40cb7b6f7e9", // clientId
                apiAppUpdateRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ApiAppApi#apiAppUpdate");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: sandbox/php/src/ApiAppUpdateExample.php
<?php

namespace Dropbox\SignSandbox;

require_once __DIR__ . '/../vendor/autoload.php';

use SplFileObject;
use Dropbox;

$config = Dropbox\Sign\Configuration::getDefaultConfiguration();
$config->setUsername("YOUR_API_KEY");
// $config->setAccessToken("YOUR_ACCESS_TOKEN");

$oauth = (new Dropbox\Sign\Model\SubOAuth())
    ->setCallbackUrl("https://example.com/oauth")
    ->setScopes([
        Dropbox\Sign\Model\SubOAuth::SCOPES_BASIC_ACCOUNT_INFO,
        Dropbox\Sign\Model\SubOAuth::SCOPES_REQUEST_SIGNATURE,
    ]);

$white_labeling_options = (new Dropbox\Sign\Model\SubWhiteLabelingOptions())
    ->setPrimaryButtonColor("#00b3e6")
    ->setPrimaryButtonTextColor("#ffffff");

$api_app_update_request = (new Dropbox\Sign\Model\ApiAppUpdateRequest())
    ->setCallbackUrl("https://example.com/dropboxsign")
    ->setName("New Name")
    ->setDomains([
        "example.com",
    ])
    ->setCustomLogoFile(new SplFileObject("CustomLogoFile.png"))
    ->setOauth($oauth)
    ->setWhiteLabelingOptions($white_labeling_options);

try {
    $response = (new Dropbox\Sign\Api\ApiAppApi(config: $config))->apiAppUpdate(
        client_id: "0dd3b823a682527788c4e40cb7b6f7e9",
        api_app_update_request: $api_app_update_request,
    );

    print_r($response);
} catch (Dropbox\Sign\ApiException $e) {
    echo "Exception when calling ApiAppApi#apiAppUpdate: {$e->getMessage()}";
}
